compiler/damlc/lib/DA/Cli/Visual.hs

```diff
@@ -125,14 +127,17 @@ constructSubgraphsWithLables lookupData tpla@TemplateChoices {..} = SubGraph nod
         nodes = map (\(chc, _) -> (chc, nodeIdForChoice lookupData chc)) choicesInTemplete
         nodesWithCreate = nodes ++ [addCreateChoice tpla lookupData]
 
-actionToChoice :: LF.Template -> Action -> LF.ChoiceName
-actionToChoice tpl (ACreate _) = LF.ChoiceName $ tplName tpl <> "_Create"
-actionToChoice tpl (AExercise _ (LF.ChoiceName "Archive")) = LF.ChoiceName $ tplName tpl <> "_Archive"
-actionToChoice _tpl (AExercise _ chc) = chc
+tplNamet :: LF.TypeConName -> T.Text
+tplNamet tplConName = head (LF.unTypeConName tplConName)
+
+actionToChoice :: Action -> LF.ChoiceName
+actionToChoice (ACreate LF.Qualified {..}) = LF.ChoiceName $ tplNamet qualObject <> "_Create"
+actionToChoice (AExercise LF.Qualified {..} (LF.ChoiceName "Archive")) = LF.ChoiceName $ tplNamet qualObject <> "_Archive"
+actionToChoice (AExercise _ chc) = chc
 
 choiceActionToChoicePairs :: ChoiceAndAction -> [(LF.ChoiceName, LF.ChoiceName)]
 choiceActionToChoicePairs cha@ChoiceAndAction {..} = pairs
-    where pairs = map (\ac -> (fst $ extractChoiceData cha, actionToChoice choiceForTemplate ac)) (Set.elems actions)
+    where pairs = map (\ac -> (fst $ extractChoiceData cha, actionToChoice ac)) (Set.elems actions)
```

Comment:
Don't think you need the `where` clause if this is the entire implementation

Reply:
Done